qemu/qmp.go

```diff
@@ -602,16 +603,30 @@ func (q *QMP) ExecuteQuit(ctx context.Context) error {
 // used to name the device.  As this identifier will be passed directly to QMP,
 // it must obey QMP's naming rules, e,g., it must start with a letter.
 func (q *QMP) ExecuteBlockdevAdd(ctx context.Context, device, blockdevID string) error {
-	args := map[string]interface{}{
-		"options": map[string]interface{}{
+	var args map[string]interface{}
+
+	if q.version.Minor <= 8 {
+		args = map[string]interface{}{
```

Comment:
Okay, having gone back and read the commit description I see there's another difference, "options".  It should still be possible to factorize the code a little though.